### PR TITLE
added an only payload endpoint for execution service

### PR DESCRIPTION
**Pull request by vivster7**
Changed queryByNamespaceNameSnapshotId to querySingle. 

I tried to minimize the amount of logic in the route itself. Personally, I think this is more readable, and we should refactor the other routes. Let me know what you think.

Also added some file level comments -- stuff I would have found helpful when I first dove into the codebase.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala [hunk 1]
```diff
@@ -17,32 +28,51 @@ abstract class AgoraService(authorizationProvider: AuthorizationProvider) extend
...
       }
     }
 
+  // Route helpers
+    val matchPath = get & path(path / Segment / Segment / IntNumber)
+    val extractOnlyParameter = extract(_.request.uri.query.get("onlyPayload"))
+
+    def extractBool(x: Option[String]): Boolean = {
+      x match {
+        case Some(x) => Try(x.toBoolean).getOrElse(false)
```
Comment (bradtaylor) [inline src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala]: Now that I get a look at this, very good use of Try. Love it.

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala [hunk 2]
```diff
@@ -17,32 +28,51 @@ abstract class AgoraService(authorizationProvider: AuthorizationProvider) extend
...
+        case None => false
+      }
+    }
```
Comment (jacarey) [inline src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala]: extract to util

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala [hunk 3]
```diff
@@ -17,32 +28,51 @@ abstract class AgoraService(authorizationProvider: AuthorizationProvider) extend
...
-        usernameFromCookie() { userName =>
+  // Route: GET http://root.com/<namespace>/<name>/<snapshotId>
+  // Route: GET http://root.com/<namespace>/<name>/<snapshotId>?onlyPayload=True
+  def querySingleRoute =
+
+  // Route logic
+    matchPath { (namespace, name, snapshotId) =>
+      usernameFromCookie() { (username) =>
+        extractOnlyParameter { (only) =>
+          val onlyBool = extractBool(only)
```
Comment (jacarey) [inline src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala]: don't need val just call extract below

**Comment by jacarey** [comment]
Looks good.

**Comment by bradtaylor** [comment]
Thumbs up!

**Pull request merged**